scalding-hadoop-test/src/test/scala/com/twitter/scalding/reducer_estimation/RuntimeReducerEstimatorTest.scala

```diff
@@ -0,0 +1,158 @@
+package com.twitter.scalding.reducer_estimation
+
+import com.twitter.scalding._
+import com.twitter.scalding.reducer_estimation.RuntimeReducerEstimator.{ RuntimePerReducer, EstimationScheme, IgnoreInputSize }
+import com.twitter.scalding.platform.{ HadoopPlatformJobTest, HadoopSharedPlatformTest }
+import org.scalatest.{ Matchers, WordSpec }
+import scala.collection.JavaConverters._
+import scala.util.{ Failure, Success, Try }
+
+object HistoryService1 extends HistoryServiceWithData {
+  import HistoryServiceWithData._
+
+  def fetchHistory(info: FlowStrategyInfo, maxHistory: Int): Try[Seq[FlowStepHistory]] =
+    Success(
+      Seq(
+        makeHistory(inputSize * 2, 0, List(10, 1000, 3000)),
+        makeHistory(inputSize / 2, 0, List(10, 200, 400)),
+        makeHistory(inputSize * 4, 0, List(10, 2400, 3000))))
+}
+
+class Estimator1 extends RuntimeReducerEstimator {
+  override val historyService = HistoryService1
```

Comment:
Probably good to also add test cases for 'empty history' and 'failed to fetch history', given we not share anything with RatioBasedEstimator so don't have coverage for that here.